src/drawtypes/label.cpp

```diff
@@ -8,7 +9,27 @@ POLYBAR_NS
 
 namespace drawtypes {
   string label::get() const {
-    return m_tokenized;
+    const size_t len = string_util::char_len(m_tokenized);
+    if (len >= m_minlen) {
+      if (m_maxlen <= 0 || len <= m_maxlen) {
+        return m_tokenized;
+      }
+      if (!m_ellipsis) {
+        return string_util::utf8_truncate(string(m_tokenized), m_maxlen);
+      }
+      return string_util::utf8_truncate(string(m_tokenized), m_maxlen - 3) + "...";
+    }
+    const size_t num_fill_chars = m_minlen - len;
+    string aligned_label;
+    if (m_alignment == label_alignment::RIGHT) {
+      aligned_label = string(num_fill_chars, ' ') + m_tokenized;
+    } else if (m_alignment == label_alignment::LEFT) {
+      aligned_label = m_tokenized + string(num_fill_chars, ' ');
+    } else {
+      const string sided_fill_chars(std::ceil(num_fill_chars / 2.0), ' ');
+      aligned_label = sided_fill_chars + m_tokenized + sided_fill_chars;
```

Comment:
for an uneven `num_fill_chars`, this would create a label that has length `m_minlen + 1`

Reply:
Yes, I wasn't sure if I should round up or down in order to respect the center alignment.
If I round down, that means the length will be `m_minlen - 1`. Do you prefer that?

Reply:
I think the following should *always* hold:
```
minlen <= len(effective label) <= maxlen
```
and if `len(m_tokenized) < minlen`, `len(effective label) == minlen`

In this case this would mean that we fill one additional space character, either on the left or on the right when `num_fill_chars` is odd. I would suggest adding it to the left side.
This could be generalized to say that the left side gets filled with `std::ceil(num_fill_chars / 2.0)` and the right side with `std::floor(num_fill_chars / 2.0)` spaces. If `num_fill_chars` is even those numbers are equal and if it's odd those numbers added together give you `num_fill_chars` again.

Reply:
> I think the following should _always_ hold:
> 
> ```
> minlen <= len(effective label) <= maxlen
> ```
I agree.
> and if `len(m_tokenized) < minlen`, `len(effective label) == minlen`
> 
I think this is more debatable. The main issue I see is that even though a user requested center alignment, they will not get it. At least in my case, I would be more surprised to see non-centered alignment than `len(m_tokenized) == minlen + 1`, assuming `minlen + 1 <= maxlen`.
Also, doing it this way means that the user can't really guarantee center alignment, since they can't restrict the labels to an even/odd length (only restrict the min/max length).
> In this case this would mean that we fill one additional space character, either on the left or on the right when `num_fill_chars` is odd. I would suggest adding it to the left side.
> This could be generalized to say that the left side gets filled with `std::ceil(num_fill_chars / 2.0)` and the right side with `std::floor(num_fill_chars / 2.0)` spaces. If `num_fill_chars` is even those numbers are equal and if it's odd those numbers added together give you `num_fill_chars` again.

Reply:
> The main issue I see is that even though a user requested center alignment, they will not get it.

Fair point. I guess as long as the case where maxlen == minlen or maxlen == minlen + 1 is handled correctly, we could do it this way.

Reply:
Just did some testing and minlen with center alignment causes some trouble.
In particular a label with text `abc`, `minlen = 4`, `maxlen = 4`, center alignmend, `maxlen = 4` and ellipsis enabled will produce ` ...` instead of `abc ` or ` abc`. I would suggest that in this special case the extra space is located at the end.

I would also suggest that we merge the functionality of this method and of `builder::get_label_text` to handle minlen, maxlen and ellipsis all in one function. Like this it's also easier to test this and how minlen and maxlen interact. We should probably move everything into `label::get` because that function is called in several places and should return the effective label in all those cases.

Reply:
Done.